examples/rust/battleship/web/client/src/game.rs

```diff
@@ -73,10 +76,120 @@ pub struct GameSession {
     pub status: String,
 }
 
+type Taken = [[bool; BOARD_SIZE];BOARD_SIZE];
+
+fn check_bounds(x: usize, y: usize, span:usize, dir:usize) -> bool {
+    let bound;
+    if dir == 0 {
+        bound = x + span <= BOARD_SIZE - 1
+    } else {
+        bound = y + span <= BOARD_SIZE - 1
+    }
+
+    // log::info!("{} {}", "check_bounds", bound);
+
+    bound
+}
```

Comment:
Since the `if` in rust is an expression, you can just do:

```suggestion
fn check_bounds(x: usize, y: usize, span:usize, dir:usize) -> bool {
    if dir == 0 {
        x + span <= BOARD_SIZE - 1
    } else {
        y + span <= BOARD_SIZE - 1
    }
}
```